### ISSUE TITLE
[question]Does just use Navigator1 ?

### ISSUE THREAD

**Issue by huang12zheng** [magicleon94/multi_page_feature#1]
Navigator2 has been around for 6 months.
although I don't understand it, 
it should be the previous mode that I find in you code

**Comment by huang12zheng** [comment]
RouteInformationParser handle state
RouterDelegate handle pop push
Page handle UI

**Comment by magicleon94** [comment]
Hi and sorry for the late reply, I didn't see the notification :(

This was an attempt to handle nested navigation with a state management approach. The first version used Navigator 2.0 but since I've stumbled upon [this problem with Navigator 2.0](https://github.com/flutter/flutter/issues/14083) I tried to change it using Navigator 1.0.

The result was an over complicated thing that I just dismissed, and forgot to delete the repo or, at least, to suggest a better solution for what I wanted to do.

The better solution is to use the [flow_builder](https://pub.dev/packages/flow_builder), which I did in several situations now in projects at work and behaves very smoothly.

Although I'm really loving this approach, the only public code I can provide for this is the one I've provided for support to [this issue](https://github.com/felangel/flow_builder/issues/10), but it's not very useful since it's based on a PR I made that got rejected.
The problem was "how do I know when a route is popped with this?"
My suggestion was adding an `onPop` callback to the `FlowBuilder` (and that's how I did in the support code), while the official solution was to provide the ability to listen on the `FlowController`.

I hope this was helpful to you! I'm planning on writing about this approach in the future but I don't know when. I'll post it here too when the article will be out :)
